### PR TITLE
Eigen: update to 3.3.1, remove old versions

### PR DESCRIPTION
**Pull request by NeroBurner**
Updating Eigen to 3.3.1, this version provides a `Eigen3Config.cmake` file itself as discussed in #623

Shall we also rename the hunter package from `Eigen` to `Eigen3` such that the `hunter_add_package` and `find_package` names match?

### CODE REVIEW

File: cmake/projects/Eigen/hunter.cmake [hunk 1]
```diff
@@ -17,26 +17,13 @@ hunter_add_version(
...
-    Eigen
-    VERSION
-    "3.2.4"
-    URL
-    "https://bitbucket.org/eigen/eigen/get/3.2.4.tar.gz"
-    SHA1
-    b826f665bdfe31784795eef79c75540db71ab702
-)
-
-hunter_cmake_args(Eigen CMAKE_ARGS EIGEN_ENABLE_TESTING=OFF)
```
Comment (ruslo) [inline cmake/projects/Eigen/hunter.cmake]: is this one removed on purpose?

**Comment by ruslo** [comment]
> Shall we also rename the hunter package from Eigen to Eigen3 such that the hunter_add_package and find_package names match?

I need to know what files installed by this new version of Eigen. Have you tried to build an example? Seems that it will not work and have to be changed too.

File: cmake/projects/Eigen/hunter.cmake (hunk 1 above)
Comment (NeroBurner) [inline cmake/projects/Eigen/hunter.cmake]: yes, the variable was introduced by you with https://github.com/hunter-packages/eigen/commit/930d0a2c836b4957b114554031c596ef219e7fb2
In Eigen 3.3.1 I was not able to find the variable `EIGEN_ENABLE_TESTING`

Comment (ruslo) [inline cmake/projects/Eigen/hunter.cmake]: > yes, the variable was introduced by you with hunter-packages/eigen@930d0a2

Heh :) Okay, I see.

> In Eigen 3.3.1 I was not able to find the variable EIGEN_ENABLE_TESTING

Also as you may noticed in comment, this variable fix the Android build https://github.com/hunter-packages/eigen/commit/930d0a2c836b4957b114554031c596ef219e7fb2

**Comment by NeroBurner** [comment]
I setup a small test project downloading the NeroBurner Eigen 3.3.1 branch
https://github.com/NeroBurner/hunter_test_eigen_download

File: cmake/projects/Eigen/hunter.cmake (hunk 1 above)
Comment (NeroBurner) [inline cmake/projects/Eigen/hunter.cmake]: > Also as you may noticed in comment, this variable fix the Android build hunter-packages/eigen@930d0a2

could you test what exactly breaks the Android build. I havn't built for Android. Maybe we can send a patch upstream to the Eigen developers which disables tests

**Comment by ruslo** [comment]
> I setup a small test project downloading the NeroBurner Eigen 3.3.1 branch

I haven't tried it but according to C++ code the header is still `Eigen/*` and not `Eigen3/*`. And C++ namespace is `Eigen::*` and not `Eigen3::*`. Hence package should be called `Eigen` and config file should be called `EigenConfig.cmake`. So let's keep the name.

File: cmake/projects/Eigen/hunter.cmake (hunk 1 above)
Comment (ruslo) [inline cmake/projects/Eigen/hunter.cmake]: Please update an example and I will run all tests, including Android.

File: examples/Eigen/CMakeLists.txt [hunk 2]
```diff
@@ -11,10 +11,10 @@ project(download-eigen)
 
 hunter_add_package(Eigen)
 
-find_package(Eigen REQUIRED)
+find_package(Eigen3 REQUIRED)
```
Comment (ruslo) [inline examples/Eigen/CMakeLists.txt]: Let it be `CONFIG REQUIRED` just to be sure

File: examples/Eigen/CMakeLists.txt [hunk 3]
```diff
@@ -11,10 +11,10 @@ project(download-eigen)
...
 
 # Test double library creation
-find_package(Eigen REQUIRED)
+find_package(Eigen3 REQUIRED)
```
Comment (ruslo) [inline examples/Eigen/CMakeLists.txt]: `CONFIG REQUIRED`

File: examples/Eigen/CMakeLists.txt (hunk 2 above)
Comment (NeroBurner) [inline examples/Eigen/CMakeLists.txt]: updated

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1250
* https://travis-ci.org/ingenue/hunter/builds/185806578

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1251
* https://travis-ci.org/ingenue/hunter/builds/185833309

**Comment by ruslo** [comment]
Released:
* https://github.com/ruslo/hunter/releases/tag/v0.17.5

Please update wiki page.

File: cmake/projects/Eigen/hunter.cmake (hunk 1 above)
Comment (ruslo) [inline cmake/projects/Eigen/hunter.cmake]: > this variable fix the Android build

Here is the case https://github.com/ruslo/hunter/issues/628

**Pull request merged**